pkg/debug/portforward.go

```diff
@@ -0,0 +1,80 @@
+package debug
+
+import (
+	"github.com/openshift/odo/pkg/occlient"
+
+	componentlabels "github.com/openshift/odo/pkg/component/labels"
+
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	k8sgenclioptions "k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+)
+
+// DefaultPortForwarder implements the SPDY based port forwarder
+type DefaultPortForwarder struct {
+	client *occlient.Client
+	k8sgenclioptions.IOStreams
+	componentName string
+	appName       string
+}
+
+func NewDefaultPortForwarder(componentName, appName string, client *occlient.Client, streams k8sgenclioptions.IOStreams) *DefaultPortForwarder {
+	return &DefaultPortForwarder{
+		client:        client,
+		IOStreams:     streams,
+		componentName: componentName,
+		appName:       appName,
+	}
+}
+
+// ForwardPorts forwards the ports using the url for the remote pod.
+// ports are list of pair of ports in format "localPort:RemotePort" that are to be forwarded
+// stop Chan is used to stop port forwarding
+// ready Chan is used to signal failure to the channel receiver
+func (f *DefaultPortForwarder) ForwardPorts(ports []string, stopChan, readyChan chan struct{}) error {
+	conf, err := f.client.KubeConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	pod, err := f.getPodUsingComponentName()
+	if err != nil {
+		return err
+	}
+
+	if pod.Status.Phase != corev1.PodRunning {
+		return fmt.Errorf("unable to forward port because pod is not running. Current status=%v", pod.Status.Phase)
```

Comment:
if the pod is not running then 
```
[mrinaldas@localhost nodejs-ex]$ odo debug port-forward
Started port forwarding at ports - 5858:5858
 ✗  unable to forward port because pod is not running. Current status=Pending
```
the output looks weird as first it says started port forwarding started and then crashes

Reply:
nice catch, resolving

Reply:
resolved